pkg/arvo/app/link-store.hoon

```diff
@@ -0,0 +1,172 @@
+::  link: social bookmarking
+::
+::    the paths under which links are submitted are generally expected to
+::    correspond to existing group paths. for strictly-local collections of
+::    links, arbitrary paths are probably fair game, but could trip up
+::    primitive ui implementations.
+::
+::  scry and subscription paths:
+::
+::    /local-pages/[some-group]    all pages we saved by recency
+::    /submissions/[some-group]    all submissions by recency
+::
+/+  *link, default-agent, verb
+::
+|%
++$  state-0
+  $:  %0
+      by-group=(map path links)
+      by-site=(map site (list [path submission]))
```

Comment:
looks good. these two types of indices / keys to query on make sense

Reply:
Note that querying by site isn't _actually_ supported yet in the scry/peer paths, but trivial to put in once we do need it. Leaving it out for now as nothing uses it.